### PR TITLE
Feature/cloud rad

### PR DESCRIPTION
**Pull request by Qingfu-Liu**
Add the convective cloud water (liquid water + ice water) in the calculations of Cloud Water Path and Ice Water Path for RRTMG radiation cloud properties. Only one file changed: radiation_clouds.f (code changed by Ruiyu Sun)
Qingfu

### CODE REVIEW

**Comment by dustinswales** [comment]
@Qingfu-Liu 
Thanks for opening this. I switched it to a "draft" for the time being. I think you may be working with an older code base?

I will propagate these RRTMG changes to RRTMGP, submit a PR to this branch, then we can start the review process.
Sound good?
(I will work on the GP code this evening)

**Comment by ChunxiZhang-NOAA** [comment]
> @Qingfu-Liu Thanks for opening this. I switched it to a "draft" for the time being. I think you may be working with an older code base?
> 
> I will propagate these RRTMG changes to RRTMGP, submit a PR to this branch, then we can start the review process. Sound good? (I will work on the GP code this evening)

@Qingfu-Liu I think your work is based on NCAR/ccpp-physics main branch. Please check out ufs-community/ccpp-physics ufs/dev branch and put your changes from there. Or you can hold there PR till ufs/dev branch is merged with the NCAR/ccpp-physics branch in about one week, then you sync your branch with ufs/dev branch.

**Comment by dustinswales** [comment]
@Qingfu-Liu @yangfanglin @JongilHan66 @RuiyuSun 
In GFS_rrtmg_pre, [before calling radiation_cloud_props](https://github.com/ufs-community/ccpp-physics/blob/ufs/dev/physics/GFS_rrtmg_pre.F90#L747), the Thompson effective particle sizes and number concentrations are updated using the cloud mixing-ratios. Since we are including the convective condensate in the cloud-fraction and water path calculations, should we add the convective condensate in here as well?

File: physics/radiation_clouds.f [hunk 1]
```diff
@@ -2216,8 +2217,12 @@ subroutine progcld_thompson_wsm6                                  &
 
         do k = 1, NLAY-1
           do i = 1, IX
-            cwp(i,k) = max(0.0, clw(i,k,ntcw) * gfac * delp(i,k))
-            cip(i,k) = max(0.0, clw(i,k,ntiw) * gfac * delp(i,k))
+            !cwp(i,k) = max(0.0, clw(i,k,ntcw) * gfac * delp(i,k))
```
Comment (dustinswales) [inline physics/radiation_clouds.f:2220]: Is it correct that the same convective condensate is added to both the liquid and ice water paths?

**Comment by yangfanglin** [comment]
bring this to @RuiyuSun attention.

**Comment by RuiyuSun** [comment]
@dustinswales In the test completed, the radii for the convective water and path are assumed to have the same as what are calculated in the calc_effectRad. What is being tested now using separate convective cloud water path, radius, and convective cloud ice path and radius.

**Comment by dustinswales** [comment]
@RuiyuSun My only concern is that the optics computed with the radiation scheme, which uses both the water paths and effective particle sizes, is now seeing water paths that has convective condensate added, while the[ calculated effective radii does not](https://github.com/ufs-community/ccpp-physics/blob/ufs/dev/physics/module_mp_thompson.F90#L5724)

**Comment by RuiyuSun** [comment]
@dustinswales This is crude first try. I hope to refine it and use separate convective cloud water path and radius at least.

**Comment by Qingfu-Liu** [comment]
@Dustin Swales ***@***.***>  In the RRTMG code changes, the
convective cloud water (partitioned to liquid water and ice water based on
local temperature) is added to the calculation of LWP and IWP. The
effective radius of liquid cloud water and ice water is assumed to be the
same as those before the code changes. I think that recalculating the
effective radius of liquid cloud water and ice water after adding the
convective cloud water is a little bit complex. We need to make assumptions
of the number concentration for convective clouds. The final effective
radius depends on the assumption, we do not know if it is better or worse
compared to the current value of effective radius.

**Comment by yangfanglin** [comment]
@dustinswales @Qingfu-Liu @RuiyuSun My suggestion is to commit this PR without using different radii and number concentrations for convective and MP cloud water and ice.   More engineering and science evaluation work needs to be done to enable that capability.  This might turned out to be necessary for us to improve cloud radiation interaction with convective cloud water being included in both cloud fraction and radiative transfer calculations. I believe Ruiyu is actively working towards testing and enabling this capability.

**Comment by dustinswales** [comment]
@yangfanglin @RuiyuSun @Qingfu-Liu Sounds like a plan.
I opened a new PR, #9, just a rebase PR from the ufs/dev fork that has both the G/GP changes included

**Pull request closed without merging**